### PR TITLE
Adding ::operatingsystem Amazon

### PR DESCRIPTION
**Pull request by dmcnaught**
We are using the Amazon Linux for a few of our servers, and needed to add it like so (no osfamily, and ::operatingsystem of Amazon) to params.pp (with the same params as RedHat).

### CODE REVIEW

**Comment by dmcnaught** [comment]
I was a little premature - I think this was fixed by a more recent version of facter, Amazon is now showing up for us as osfamily RedHat, so the module works. Sorry for any inconvenience.

**Comment by dmcnaught** [comment]
I've looked at this more - and we would like this code added - as it would support the version of facter that gets installed with cloud-init onto our Amazon systems.

**Comment by elyscape** [comment]
Which version of Facter is included on those systems?  Also, a way to avoid duplication of the parameters would be to instead change [line 90](https://github.com/dmcnaught/puppetlabs-ntp/blob/master/manifests/params.pp#L90) to `'RedHat', 'Amazon': {`.

**Comment by dmcnaught** [comment]
facter-1.6.18-7.26.amzn1.noarch 
from
facter.noarch                                        1.6.18-7.26.amzn1                                        @amzn-main

**Comment by dmcnaught** [comment]
Thanks for your message Eli.
Facter version is facter-1.6.18-7.26.amzn1.noarch
from facter.noarch 1.6.18-7.26.amzn1 @amzn-main
The osfamily fact comes up as Linux - which is why I added it to that
section.


Thanks
--Duncan
____________________________

Duncan McNaught
Infrastructure Engineer
 Technologies | www.bitnet.io
+1 720 240 6575

**Comment by elyscape** [comment]
Interesting.

Looks like this was fixed [in Facter 1.7.0](https://github.com/puppetlabs/facter/commit/c12d3b6c557df695a7b2b009da099f6a93c7bd31).  I wonder why Amazon installs 1.6.18.  Any ideas?

**Comment by dmcnaught** [comment]
Looks like it's to do with Ruby 1.8:
http://aws.amazon.com/amazon-linux-ami/2015.03-release-notes/

**Comment by elyscape** [comment]
That could do it.

**Comment by elyscape** [comment]
Do new provisionings using the newest release of Amazon Linux still install Facter 1.6.18 or do they use Facter 2?

**Comment by dmcnaught** [comment]
Doing a yum install puppet will get Facter 1.6.18.
Doing yum install puppet3 will get Facter2 from the amazon repo, which is facter2-2.2.0-1.7.amzn1.x86_64

**Comment by elyscape** [comment]
Looks like cloud-init will use the old Puppet, too.  How frustrating.

**Comment by dmcnaught** [comment]
Can we merge this pull request? Is there something else I should do?

**Comment by jonnytpuppet** [comment]
Hi There

You need to update the metadata.json file with the OS update, as well as update the readme to say that this module is 'compatible' with Amazon.

Thanks

**Comment by dmcnaught** [comment]
Thanks jonnytpuppet, I've added the relevant Amazon section to metadata.json.
I didn't update the readme, as it doesn't reference any of the Linux versions the module is compatible with.

**Comment by dmcnaught** [comment]
Those Travis failures don't seem related to my change at all.

File: metadata.json [hunk 1]
```diff
@@ -87,6 +87,15 @@
         "7.1"
       ]
     }
```
Comment (adrienthebo) [inline metadata.json]: Missing a comma

Comment (dmcnaught) [inline metadata.json]: Thanks Adrien

**Comment by dmcnaught** [comment]
Can we remove the "needs work" label now?

**Comment by elyscape** [comment]
Please rebase the branch and then squash the commits down.

**Comment by dmcnaught** [comment]
Thanks Eli. I think that's it. What happened to Travis?

File: metadata.json [hunk 2]
```diff
@@ -98,7 +107,7 @@
       "version_requirement": "3.x"
     }
   ],
-  "description": "NTP Module for Debian, Ubuntu, CentOS, RHEL, OEL, Fedora, FreeBSD, ArchLinux and Gentoo.",
+  "description": "NTP Module for Debian, Ubuntu, CentOS, RHEL, OEL, Fedora, FreeBSD, ArchLinux, Amazon and Gentoo.",
```
Comment (elyscape) [inline metadata.json]: This should probably say "Amazon Linux" instead of "Amazon".

**Comment by dmcnaught** [comment]
ok, thanks Eli. I've done that too. :-)

File: metadata.json (hunk 2 above)
Comment (dmcnaught) [inline metadata.json]: ok, thanks Eli. I've done that too. :-)

File: manifests/params.pp [hunk 3]
```diff
@@ -283,6 +283,46 @@
           $maxpoll         = undef
           $disable_monitor = false
         }
+        'Amazon': {
```
Comment (elyscape) [inline manifests/params.pp]: Looks like this section got duplicated.

Comment (dmcnaught) [inline manifests/params.pp]: How weird. My local copy didn't have the duplication and showed no diffs/pull. Sorry. Updating now.

Comment (dmcnaught) [inline manifests/params.pp]: Completed.

File: manifests/params.pp [hunk 4]
```diff
@@ -283,6 +283,26 @@
...
+            '127.0.0.1',
+            '-6 ::1',
+          ]
+          $iburst_enable   = false
+          $servers         = [
+            '0.centos.pool.ntp.org',
+            '1.centos.pool.ntp.org',
+            '2.centos.pool.ntp.org',
+          ]
+          $maxpoll         = undef
```
Comment (mhaskel) [inline manifests/params.pp]: You're missing the `$disable_monitor` parameter.

Comment (dmcnaught) [inline manifests/params.pp]: Yeah - that was refactored on June 15th - 2 days after my last change.
Why does it take so long for any movement on this pull request. There will always be work to do if we wait for work to happen around it...

**Comment by dmcnaught** [comment]
I'm not sure why I can't pull, merge and rebase here, but I've been going around in circles.
I'm going to re-fork and add my simple change again.

**Comment by dmcnaught** [comment]
Reforked to https://github.com/puppetlabs/puppetlabs-ntp/pull/291

**Pull request closed without merging**